src/clj/game/cards/events.clj

```diff
@@ -215,11 +215,14 @@
                                     c)))})
 
    "Contaminate"
-   {:msg (msg "place 3 virus tokens on " (:title target))
-    :choices {:req #(and (installed? %)
-                         (= (:side %) "Runner")
-                         (= 0 (get-in % [:counter :virus] 0)))}
-    :effect (req (add-counter state :runner target :virus 3))}
+   {:effect (req (resolve-ability
+                   state side
+                   {:msg (msg "place 3 virus tokens on " (:title target))
+                    :choices {:req #(and (installed? %)
+                                         (= (:side %) "Runner")
+                                         (= 0 (get-virus-counters state side %)))}
+                    :effect (req (add-counter state :runner target :virus 3))}
```

Comment:
Should be able to use `effect` here instead of `req` if you so fancy. I personally do not mind.